src/wing/CCPACSWingCell.cpp

```diff
@@ -351,46 +348,549 @@ std::pair<double, double> CCPACSWingCell::computePositioningEtaXsi(const CCPACSW
 
         tigl::EtaXsi result = computeRibSparIntersectionEtaXsi(wsr, ribsDefinition, ribIndex, spar);
-        eta = result.eta;
-        xsi = result.xsi;
+        eta                 = result.eta;
+        xsi                 = result.xsi;
     }
 
     return std::make_pair(eta, xsi);
 }
 
 void CCPACSWingCell::UpdateEtaXsiValues() const
 {
-    cache.emplace();
+    cache.valid = false;
+
+    std::pair<double, double> innerLePoint =
+        computePositioningEtaXsi(m_positioningInnerBorder, m_positioningLeadingEdge, true, true);
+    cache.innerLeadingEdgePoint.eta = innerLePoint.first;
+    cache.innerLeadingEdgePoint.xsi = innerLePoint.second;
 
-    std::pair<double, double> innerLePoint = computePositioningEtaXsi(m_positioningInnerBorder, m_positioningLeadingEdge, true, true);
-    cache->innerLeadingEdgePoint.eta = innerLePoint.first;
-    cache->innerLeadingEdgePoint.xsi = innerLePoint.second;
+    std::pair<double, double> outerLePoint =
+        computePositioningEtaXsi(m_positioningOuterBorder, m_positioningLeadingEdge, false, true);
+    cache.outerLeadingEdgePoint.eta = outerLePoint.first;
+    cache.outerLeadingEdgePoint.xsi = outerLePoint.second;
 
-    std::pair<double, double> outerLePoint = computePositioningEtaXsi(m_positioningOuterBorder, m_positioningLeadingEdge, false, true);
-    cache->outerLeadingEdgePoint.eta = outerLePoint.first;
-    cache->outerLeadingEdgePoint.xsi = outerLePoint.second;
+    std::pair<double, double> innerTePoint =
+        computePositioningEtaXsi(m_positioningInnerBorder, m_positioningTrailingEdge, true, false);
+    cache.innerTrailingEdgePoint.eta = innerTePoint.first;
+    cache.innerTrailingEdgePoint.xsi = innerTePoint.second;
 
-    std::pair<double, double> innerTePoint = computePositioningEtaXsi(m_positioningInnerBorder, m_positioningTrailingEdge, true, false);
-    cache->innerTrailingEdgePoint.eta = innerTePoint.first;
-    cache->innerTrailingEdgePoint.xsi = innerTePoint.second;
+    std::pair<double, double> outerTePoint =
+        computePositioningEtaXsi(m_positioningOuterBorder, m_positioningTrailingEdge, false, false);
+    cache.outerTrailingEdgePoint.eta = outerTePoint.first;
+    cache.outerTrailingEdgePoint.xsi = outerTePoint.second;
 
-    std::pair<double, double> outerTePoint = computePositioningEtaXsi(m_positioningOuterBorder, m_positioningTrailingEdge, false, false);
-    cache->outerTrailingEdgePoint.eta = outerTePoint.first;
-    cache->outerTrailingEdgePoint.xsi = outerTePoint.second;
+    cache.valid = true;
 }
 
 void CCPACSWingCell::Update() const
 {
-    // TODO: update geometry
+    if (geometryValid) {
+        return;
+    }
+
+    BuildSkinGeometry();
+    geometryValid = true;
+}
+
+TopoDS_Shape CCPACSWingCell::GetCellSkinGeometry(bool transform) const
+{
+    Update();
+
+    if (transform) {
+        return m_parent->GetParentElement()
+            ->GetStructure()
+            .GetWingStructureReference()
+            .GetWingComponentSegment()
```

Comment:
this works only if the parent transitively is a component segment.

Reply:
What do you suggest?

Reply:
I now used the uid manager for it.